src/components/TrafficList/TrafficListComponent.tsx

```diff
@@ -0,0 +1,353 @@
+  }
+
+  // abstract FilterComponent.updateListItems
+  updateListItems() {
+    // we don't react to filter changes in this class, so this is a no-op
+  }
+
+  // abstract FilterComponent.sortItemList
+  sortItemList(
+    listItems: TrafficListItem[],
+    sortField: SortField<TrafficListItem>,
+    isAscending: boolean
+  ): Promise<TrafficListItem[]> {
+    const sorted = listItems.sort(isAscending ? sortField.compare : (a, b) => sortField.compare(b, a));
+    return Promise.resolve(sorted);
+  }
+
+  // Helper used for Table to sort handlers based on index column == field
+  onSort = (_event, index, sortDirection) => {
+    // Map the column index to the correct sortField index (currently ordered with the same indexes)
+    let sortField = sortFields[index];
+
+    const isSortAscending = sortDirection === SortByDirection.asc;
+    if (sortField.id !== this.state.currentSortField.id) {
+      this.updateSortField(sortField);
+    } else if (isSortAscending !== this.state.isSortAscending) {
+      this.updateSortDirection();
+    }
+
+    this.setState({ sortBy: { index: index, direction: sortDirection } });
+  };
+
+  trafficToListItems(trafficItems: TrafficItem[]) {
+    const listItems = trafficItems.map(ti => {
+      let icon: string;
+      switch (ti.node.type) {
+        case NodeType.APP:
+          icon = 'A';
+          break;
+        case NodeType.SERVICE:
+          icon = 'S';
+          break;
+        default:
+          icon = 'W';
+      }
+      const item: TrafficListItem = {
+        icon: icon,
+        node: ti.node,
+        protocol: this.getProtocol(ti.traffic),
+        healthStatus: this.getHealthStatus(ti),
+        ...this.getTraffic(ti.traffic)
+      };
+      return item;
+    });
+
+    return listItems;
+  }
+
+  private getHealthStatus = (item: TrafficItem): ThresholdStatus => {
+    const traffic = item.traffic;
+
+    if (traffic.protocol !== 'tcp' && hasProtocolTraffic(traffic)) {
+      return getTrafficHealth(item, this.props.direction);
+    }
+
+    return { value: 0, status: NA };
+  };
+
+  private getTraffic = (traffic: ProtocolTraffic): { trafficRate; trafficPercent } => {
+    let rps = '0';
+    let percentError = '0';
+    let unit = 'rps';
+    if (hasProtocolTraffic(traffic)) {
+      switch (traffic.protocol) {
+        case 'http':
+          rps = traffic.rates.http;
+          percentError = traffic.rates.httpPercentErr || '0';
+          break;
+        case 'grpc':
+          rps = traffic.rates.grpc;
+          percentError = traffic.rates.grpcPercentErr || '0';
+          break;
+        case 'tcp':
+          rps = traffic.rates.tcp;
+          break;
+      }
+    }
+
+    return {
+      trafficRate: `${Number(rps).toFixed(2)}${unit}`,
+      trafficPercent: `${(100 - Number(percentError)).toFixed(1)}%`
+    };
+  };
+
+  private getProtocol = (traffic: ProtocolTraffic) => {
+    if (!traffic.protocol) {
+      return 'N/A';
+    }
+
+    return traffic.protocol.toUpperCase();
```

Comment:
```typescript
private getProtocol = (traffic) => (traffic.protocol || "N/A").toUpperCase();
```

Reply:
Yours is prettier, I admit.  I don't encourage unsolicited style reviews but I'll throw you a bone and apply this one.  I'll even do you one better and remove the single-use function completely.